packages/venia-concept/src/components/CmsBlock/cmsBlock.js

```diff
@@ -0,0 +1,57 @@
+import { Component, createElement } from 'react';
+import { PropTypes } from 'prop-types';
+import { Query } from 'react-apollo';
+import gql from 'graphql-tag';
+import classify from 'src/classify';
+import defaultClasses from './cmsBlock.css';
+
+const cmsBlockQuery = gql`
+    query cmsBlocks($identifiers: [String]!) {
+        cmsBlocks(identifiers: $identifiers) {
+            items {
+                content
+            }
+        }
+    }
+`;
+
+class CmsBlock extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            root: PropTypes.string,
+            content: PropTypes.string
+        }),
+        identifiers: PropTypes.oneOfType([PropTypes.string, PropTypes.array])
+    };
+
+    render() {
+        const { identifiers, classes } = this.props;
+        return (
+            <div className={classes.root}>
+                <Query query={cmsBlockQuery} variables={{ identifiers }}>
+                    {({ loading, error, data }) => {
+                        if (error) return <div>Data Fetch Error</div>;
+                        if (loading) return <div>Fetching Data</div>;
+                        if (data.cmsBlocks.items == '')
+                            return <div>Here is not any block(s)</div>;
+
+                        return (
+                            <div className={classes.content}>
+                                {data.cmsBlocks.items.map((item, index) => (
+                                    <div
```

Comment:
I think that you should delegate to a function-as-child-component here. The arguments to the child component could be a div containing the block content, the identifier, and the title. That way, someone could use different iterative rendering strategies for different collections of components.

As it is, I have to use a CMS block like this:

```jsx
<CmsBlock identifiers={["block1", "block2", "block3"]} />
```

And this renders:

```html
<div class="root">
  <div class="content">
    <div>Content 1</div>
    <div>Content 2</div>
    <div>Content 3</div>
  </div>
</div>
```

That's fine as default behavior _(though the inner divs should have a classname!)_ but I would also like to be able to do something like this:

```jsx
<div>
  <h2>Quick Links</h2>
  <ul>
    <CmsBlock identifiers={['block1','block2','block3']}>
      {({ key, identifier, title }) => (
        <li key={key}>
          <a href={`#anchor-${identifier}`}>{title}</a>
        </li>
      )}
    </CmsBlock>
  </ul>
  <h2>Sections</h2>
  <CmsBlock identifiers={['block1','block2','block3']}>
    {({ key, identifier, title, content }) => (
      <div key={key} id={`anchor-${identifier}`}>
        <h3>{title}</h3>
        <div dangerouslySetInnerHTML={{ __html: item.content }}></div>
      </div>
    )}
  </CmsBlock>
</div>
```

Which would yield this:

```html
<div>
  <h2>Quick Links</h2>
  <div class="root">
    <ul>
      <li>
        <a href="#anchor-block1">Title 1</a>
      </li>
      <li>
        <a href="#anchor-block2">Title 2</a>
      </li>
      <li>
        <a href="#anchor-block3">Title 3</a>
      </li>
    </ul>
  </div>
  <h2>Sections</h2>
  <div id="anchor-block1">
    <h3>Title 1</h3>
    <div>Content 1</div>
  </div>
  <div id="anchor-block2">
    <h3>Title 2</h3>
    <div>Content 2</div>
  </div>
  <div id="anchor-block3">
    <h3>Title 3</h3>
    <div>Content 3</div>
  </div>
</div>
```

So your implementation here would just assume that `children` was a function (specified in `propTypes` of course), and do something like this:

```jsx
const renderBlock = this.props.children || defaultImpl;
return (
  <div className={classes.content}>
    {data.cmsBlocks.items.map((item, index) => renderBlock({ key: index, ...item }))}
  </div>
);
```

Reply:
That is awesome idea!
I like it.